cabal-install/src/Distribution/Client/CmdUpdate.hs

```diff
@@ -197,10 +197,12 @@ updateRepo verbosity _updateFlags repoCtxt (repo, indexState) = do
       -- NB: always update the timestamp, even if we didn't actually
       -- download anything
       writeIndexTimestamp index indexState
-      ce <- if repoContextIgnoreExpiry repoCtxt
-              then Just `fmap` getCurrentTime
-              else return Nothing
-      updated <- Sec.uncheckClientErrors $ Sec.checkForUpdates repoSecure ce
+      -- typically we get the current time to check expiry against
+      -- but if the flag is set, we don't.
+      now <- case repoContextIgnoreExpiry repoCtxt of
+                 False -> Just <$> getCurrentTime
+                 True  -> pure Nothing
+      updated <- Sec.uncheckClientErrors $ Sec.checkForUpdates repoSecure now
```

Comment:
```suggestion
      updated <- do
        now <- case repoContextIgnoreExpiry repoCtxt of
                   False -> Just <$> getCurrentTime
                   True  -> pure Nothing
        Sec.uncheckClientErrors $ Sec.checkForUpdates repoSecure now
```

Reply:
Ah, it seems it was later fixed differently: https://github.com/haskell/cabal/commit/ec86ddcdd990b358ada9395c41a94a408c5eb9a8

No idea why the commit messages are unhelpful.

Reply:
Yeah, commit messages and PRs left me in the dark. I used the same version as in https://github.com/haskell/cabal/commit/ec86ddcdd990b358ada9395c41a94a408c5eb9a8 just to keep things consistent. Thank you.